src/lib/query/vlan.rs

```diff
@@ -10,23 +10,21 @@ use crate::{Iface, IfaceType};
 #[derive(
     Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy, Default,
 )]
-#[serde(rename_all = "snake_case")]
-#[non_exhaustive]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
 pub enum VlanProtocol {
     #[serde(rename = "802.1q")]
     #[default]
     Ieee8021Q,
     #[serde(rename = "802.1ad")]
-    Ieee8021AD,
-    Unknown,
+    Ieee8021Ad,
```

Comment:
Need reason why break API.

Reply:
To simplify things in nmstate given that this is very stable (unlikely to have new protocols).

As you know, we need to have something here because of the match in `from`.  The compiler complains that the struct `VlanProtocol` of `netlink-packet-route` is non-exhaustive (something could be added - unlikely). So, if I wouldn't changed this,  I would have had to do the same for the VlanProtocol structure in nmstate because I'm also using a `from` there. The other alternative is to return an error in `gen_sriov_conf`, similar to what is done for Linux Bridge in `nisport/linux_bridge.rs` but that requires to change the return type of that function.

So, I figured we could simplify this given that it's very unlikely to add a new parameter for `VlanProtocol` in `netlink-packet-route`. This simplifies a lot the logic in nmstate. I know that strictly speaking the panic in the `todo` shouldn't be in production code but this is very stable.

Reply:
alternative, could we remove the non-exhaustive for `VlanProtocol` of `netlink-packet-route`?

Regarding the AD to Ad change, it was to align the same name that it is present in nmstate. Since I didn't see logic to apply (set) changes for sriov here in this project. I thought it was safe to do it. Please correct me if I'm wrong.